modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/enricher/DisplayModelEnricher.java

```diff
@@ -0,0 +1,150 @@
+import org.mule.runtime.module.extension.internal.model.property.ImplementingMethodModelProperty;
+import org.mule.runtime.module.extension.internal.model.property.ImplementingParameterModelProperty;
+import org.mule.runtime.module.extension.internal.model.property.ImplementingTypeModelProperty;
+import org.mule.runtime.module.extension.internal.util.IdempotentDeclarationWalker;
+import org.mule.runtime.module.extension.internal.util.IntrospectionUtils;
+
+import java.lang.reflect.AnnotatedElement;
+import java.util.Optional;
+
+/**
+ * Enriches the {@link ExtensionDeclarer} with a {@link DisplayModelProperty} from annotated elements with
+ * {@link Summary} or {@link DisplayName}
+ *
+ * @since 4.0
+ */
+public final class DisplayModelEnricher extends AbstractAnnotatedModelEnricher
+{
+
+    @Override
+    public void enrich(DescribingContext describingContext)
+    {
+        new IdempotentDeclarationWalker()
+        {
+            @Override
+            public void onSource(WithSourcesDeclaration owner, SourceDeclaration declaration)
+            {
+                enrichTypes(declaration);
+                super.onSource(owner, declaration);
+            }
+
+            @Override
+            public void onParameter(ParameterizedInterceptableDeclaration owner, ParameterDeclaration declaration)
+            {
+                enrichParameter(declaration);
+                super.onParameter(owner, declaration);
+            }
+
+            @Override
+            public void onOperation(WithOperationsDeclaration owner, OperationDeclaration declaration)
+            {
+                enrichOperation(declaration);
+                super.onOperation(owner, declaration);
+            }
+
+            @Override
+            public void onConnectionProvider(ConnectedDeclaration owner, ConnectionProviderDeclaration declaration)
+            {
+                enrichTypes(declaration);
+                super.onConnectionProvider(owner, declaration);
+            }
+
+            @Override
+            public void onConfiguration(ConfigurationDeclaration declaration)
+            {
+                enrichTypes(declaration);
+                super.onConfiguration(declaration);
+            }
+        }.walk(describingContext.getExtensionDeclarer().getDeclaration());
+    }
+
+    private void enrichParameter(ParameterDeclaration declaration)
+    {
+        final Optional<DeclaringMemberModelProperty> declaringMemberProperty = declaration.getModelProperty(DeclaringMemberModelProperty.class);
+        final Optional<ImplementingParameterModelProperty> implementingParameterProperty = declaration.getModelProperty(ImplementingParameterModelProperty.class);
+        AnnotatedElement annotatedElement = null;
+
+        if (declaringMemberProperty.isPresent())
+        {
+            annotatedElement = declaringMemberProperty.get().getDeclaringField();
+        }
+
+        if (implementingParameterProperty.isPresent())
```

Comment:
although it can be assumed, the lack of an annotation does not imply the existence of the other